### PR TITLE
Create proper isolated cache for tx index in transaction verifier

### PR DESCRIPTION
**Pull request by TheQuantumPhysicist**

### CODE REVIEW

File: chainstate/src/detail/transaction_verifier/tx_index_cache.rs [hunk 1]
```diff
@@ -0,0 +1,187 @@
+use std::collections::{btree_map::Entry, BTreeMap};
```
Comment (azarovh) [inline chainstate/src/detail/transaction_verifier/tx_index_cache.rs:1]: copyright

**Review by azarovh — APPROVED** [review]

File: chainstate/src/detail/transaction_verifier/tx_index_cache.rs (hunk 1 above)
Comment (TheQuantumPhysicist) [reply chainstate/src/detail/transaction_verifier/tx_index_cache.rs:1]: Oh, crap...

Comment (TheQuantumPhysicist) [reply chainstate/src/detail/transaction_verifier/tx_index_cache.rs:1]: I wonder why the python script didn't figure it out

**Review by stanislav-tkach — APPROVED** [review]

**Review by muursh — APPROVED** [review]
lgtm

**Review by iljakuklic — APPROVED** [review]

**Pull request merged**